crates/telio-nurse/src/qos.rs

```diff
@@ -181,9 +199,16 @@ impl Runtime for Analytics {
                 })
             },
 
-            _ = self.rtt_interval.tick() => {
+            _ = self.rtt_interval.tick(), if self.ping_channel_tx.upgrade().is_none() => {
+                Self::guard(async move {
+                    self.perform_ping();
+                    Ok(())
+                })
+            },
+
+            Some(node_ping_res) = self.ping_channel_rx.recv() => {
                 Self::guard(async move {
-                    self.perform_ping().await;
+                    self.process_node_ping_results(node_ping_res);
```

Comment:
that's correct

Reply:
> mostly because I don't 100% understand why the issue is sporadic.

Pinging invalid endpoints would block the task on perform_ping(), since it's guarded it wouldn't be interrupted..

Reading you're question about the guard I now realize that probably removing the guard would help.. wdyt?

Reply:
Yup, I think that the guard is not really needed.